src/extension/src/utils/dependencyChecker.ts

```diff
@@ -0,0 +1,79 @@
+import { WizardServant, IPayloadResponse } from "../wizardServant";
+import { ExtensionCommand, CONSTANTS } from "../constants";
+const os = require('os');
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+
+const PYTHON3_REGEX = RegExp('^Python 3\\.[5-9]\\.[0-9]');
+
+export class DependencyChecker extends WizardServant {
+  clientCommandMap: Map<
+    ExtensionCommand,
+    (message: any) => Promise<IPayloadResponse>
+  >;
+
+  constructor() {
+    super();
+    this.clientCommandMap = this.defineCommandMap();
+  }
+
+  private defineCommandMap(): Map<
+    ExtensionCommand,
+    (message: any) => Promise<IPayloadResponse>
+  > {
+    return new Map([[ExtensionCommand.CheckDependency, this.checkDependency]]);
+  }
+
+  private async runPythonVersionCommand(command: string) {
+    let installed: boolean;
+    try {
+      const { stdout } = await exec(command + ' --version');
+      if (PYTHON3_REGEX.test(stdout)) {
+        installed = true;
+      } else {
+        installed = false;
+      }
+    } catch (err) {
+      installed = false;
+    }
+    return installed;
+  }
+
+  async checkDependency(message: any): Promise<IPayloadResponse> {
+    let name: string = message.payload.dependency;
+    let state: boolean = false;
+    if (name === CONSTANTS.DEPENDENCY_CHECKER.NODE) {
+      try { 
+        const { stdout } = await exec(CONSTANTS.DEPENDENCY_CHECKER.NODE + ' --version');
+        if (stdout.length > 0) { 
+          state = true;
+        } else {
+          state = false;
+        }
+      } catch (err) {
+        state = false;
```

Comment:
Is swallowing the error okay here? Do you think we should be logging this (now or later)?

Reply:
@SahilTara We could add logging here, but if an error is throw, it means that node/ python is not installed, and since we are only showing 2 states on the client side, I thought swallowing the error is ok. 

Unless we want to give the user more information, then I will be happy to add logging.

Reply:
Ah okay makes sense. I wasn't sure when the error was thrown so I wanted to ask.